### DISCUSSION TITLE
Not able to fetch data after deployment due to data cache

### DISCUSSION THREAD

**Discussion by imvishalvyas** [vercel/community#6852] [Help]
### Summary

I deployed the app in the vercel, and my website retrieves data from the sanity web. However, my website does not retrieve new changes from sanity after any changes occur, and I am also running a new deployment after sanity is updated. It only shows the most recent data after purging the data cache from the Vercel console. When any data in the Sanity database is updated, I must purge the Vercel console. If anyone has experienced the same problem, please advise me on how to resolve it.

### Steps to Reproduce

- Update data in sanity dataset.
- Deploy vercel website
- changes not showing
- Purge the data cache
- Showing changes

**Comment by jososadev** [comment]
I'm having the same issue. Seems that if you run it locally as `npm run build && npm run start` the same problem happens.

  I think it's related to Next instead of Vercel.

**Comment by amyegan** [comment]
I don't know enough about your app setup to say for sure, but it sounds like you may need to configure some version of [Incremental Static Regeneration (ISR)](https://vercel.com/docs/incremental-static-regeneration#). The [Blog with Native Content Editing](https://vercel.com/templates/next.js/blog-nextjs-sanity) has an example with its [`revalidate.ts` file](https://github.com/sanity-io/nextjs-blog-cms-sanity-v3/blob/d50c905fb43c416ca5b5879960dfda6533c032dd/pages/api/revalidate.ts#L47).

If you're using the Next.js App Router you can find more info about data fetching, caching, and revalidating at [nextjs.org/docs/app/building-your-application/data-fetching/fetching-caching-and-revalidating](https://nextjs.org/docs/app/building-your-application/data-fetching/fetching-caching-and-revalidating)

**Reply by versecafe** [reply]
On this note, when using App Router you can always have in an admin panel with an action or API endpoint that just calls something like this to just manually invalidate the cache on all of your relevant routes *and tags if using `unstable_cache`*

```ts
"use server";

import { revalidatePath, revalidateTag } from "next/cache";

export async function revalidateAll(): Promise<void> {
  await revalidateArticles();
  await revalidateCommunities();
  await revalidateTests();
  await revalidateLanders();
}

export async function revalidateArticles(): Promise<void> {
  revalidateTag("articles");
  revalidatePath("/articles", "page");
  revalidatePath("/articles/[slug]", "page");
  console.log("revalidating articles");
}

export async function revalidateCommunities(): Promise<void> {
  revalidateTag("communities");
  revalidatePath("/communities", "page");
  revalidatePath("/communities/[slug]", "page");
  console.log("revalidating communities");
}

export async function revalidateTests(): Promise<void> {
  revalidateTag("tests");
  revalidatePath("/tests", "page");
  revalidatePath("/tests/[slug]", "page");
  console.log("revalidating tests");
}

export async function revalidateLanders(): Promise<void> {
  revalidateTag("landers");
  revalidatePath("/landers/[slug]", "page");
  console.log("revalidating landers");
}

```

> *pulled from a production example*

**Reply by shubham-tarpara** [reply]
I'm facing the same issue have created a sanity webhook for it and in nextjs created API router to revalidate it
```
import { NextResponse } from "next/server";
import { revalidatePath } from "next/cache";
import { getAllSlugs } from "../../../sanity/queries/queries";

export async function POST(req) {
  const secret = process.env.SANITY_WEBHOOK_SECRET;
  if (req.headers["sanity-webhook-signature"] !== secret) {
    return NextResponse.json({ message: "Unauthorized" }, { status: 401 });
  }
  try {
    const slugs = await getAllSlugs();
    for (const slug of slugs) {
      if (typeof slug === "string") {
        revalidatePath(slug);
      }
    }
    return NextResponse.json({ message: "Revalidated all pages" });
  } catch (error) {
    console.error(error);
    return NextResponse.json(
      { message: "Error revalidating" },
      { status: 500 }
    );
  }
}
```

It is still not working and one more thing if I use 
export const fetchCache = "force-no-store";
Will this make my website not appear for a second showing a network error? and after hard refresh it will work fine?
